src/flashbar/collapsible.scss

```diff
@@ -0,0 +1,229 @@
+       with one single line of text.
+       This makes them look better during the collapse animation, in which they are already empty.
+     */
+    min-height: calc(
+      #{awsui.$font-body-m-line-height} + (
+          #{awsui.$space-scaled-xs} + #{awsui.$border-field-width} + #{awsui.$space-scaled-xxs}
+        ) * 2
+    );
+  }
+
+  &.visual-refresh {
+    > .item:not(:last-child) > .flash,
+    > .item.flash {
+      box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
+    }
+  }
+
+  &:not(.visual-refresh) {
+    > .item:not(:last-child) > .flash,
+    > .item.flash {
+      box-shadow: 0px 2px 2px rgba(0, 0, 0, 0.15);
+    }
+  }
+}
+
+.stack > .expanded.animation-running > .flash-list-item {
+  position: relative;
+  @include stacked-flash-z-index;
+}
+
+@mixin toggle-button-layout($margin-top-when-collapsed, $border-width) {
+  $line-height: awsui.$font-body-m-line-height;
+  $padding-vertical: styles.$control-padding-vertical;
+  $margin-top-when-expanded: -12px;
+
+  padding-top: $padding-vertical;
+  padding-bottom: $padding-vertical;
+
+  &:not(.expanded) {
+    margin-top: #{$margin-top-when-collapsed};
+    margin-bottom: calc(
+      #{-$margin-top-when-collapsed} - (#{$line-height} + 2 * (#{$padding-vertical} + #{$border-width}))
+    );
+  }
+
+  &.expanded {
+    margin-top: #{$margin-top-when-expanded};
+    margin-bottom: calc(
+      #{-$margin-top-when-expanded} - (#{$line-height} + 2 * (#{$padding-vertical} + #{$border-width}))
+    );
+  }
+}
+
+.stack > .toggle {
+  @include styles.text-wrapping;
+
+  background: awsui.$color-background-layout-toggle-default;
+  border-color: awsui.$color-background-layout-toggle-default;
+  border-radius: awsui.$border-radius-button;
+  border-style: solid;
+  box-shadow: awsui.$shadow-panel-toggle;
+  color: awsui.$color-text-layout-toggle;
+  cursor: pointer;
+  display: flex;
+  grid-column: 2;
+  grid-row: 2;
+  gap: awsui.$space-xs;
+  letter-spacing: awsui.$font-button-letter-spacing;
+  margin-left: auto;
+  margin-right: auto;
+  text-decoration: none;
+  z-index: 1;
+
+  > .status > .text,
+  > .status > .types-count,
+  > .button {
+    @include typography.default-text-style;
+    color: awsui.$color-text-layout-toggle;
+    cursor: pointer;
+    margin-top: 0;
+    margin-bottom: 0;
+  }
+
+  > .status {
+    > .text {
+      font-weight: awsui.$font-button-weight;
+      display: inline;
+      margin-right: awsui.$space-s;
+    }
+
+    > .types-count {
+      margin-left: awsui.$space-xxs;
+
+      > .types-count-visible {
+        > .type-count {
+          margin-right: awsui.$space-s;
+
+          > .count-number {
+            margin-left: awsui.$space-xxs;
+          }
+        }
+      }
+    }
+  }
+
+  &:hover {
+    background: awsui.$color-background-layout-toggle-hover;
+  }
+
+  &:active {
+    background: awsui.$color-background-layout-toggle-active;
+  }
+
+  &.visual-refresh {
+    $border-width: 2px;
+    @include toggle-button-layout(-27px, $border-width);
```

Comment:
What are these magic numbers representing? Can we calculate them or derive them from something?

Reply:
This is the amount of space that the expand/collapse button is pushed up to overlap the Flashbar. I just based this on the given UI design specification. I know it's a very arbitrary number but it's also a very specific case and the number doesn't really come from anything else. I guess I could approximate it to use a combination of tokens if needed —would that be preferable?

Reply:
If it can't be calculated properly, that's fine. But then you could at least put it in a SASS variable and/or add a comment.

Reply:
OK, done both things (SASS variable and comment) in latest revision